src/doc/threat_model/threat_model.md

```diff
@@ -151,17 +151,20 @@ and its security-relevant dependencies:
 
 ## What is an "Application"?
 
-Tock does not currently have a precise definition of "application", although
-there is consensus on the following:
-
-- Unlike a process, an application persists across reboots and updates. For
-  example, an application binary can be updated without becoming a new
-  application but the update will create a new process.
-
-- An application consists of at least one application binary (in the Tock Binary
-  Format), although it is unclear whether multiple application binaries can
-  collectively be considered a single application (e.g. if they implement a
-  single piece of functionality).
-
-This section will be updated when we have a more precise definition of
-"application".
+Formally, a Tock application is the set of all processes that have a particular
+application ID (as detailed in
+[the AppID TRD](https://github.com/tock/tock/blob/master/doc/reference/trd-appid.md)).
+
+Every process has an application ID (which may be global or locally unique), so
+every process is part of an application.
+
+Applications are generally used as a way to grant access to something. For
+example, a process that wants to send a message to another process will
+generally do so by sending the message to that process' application ID. Doing so
+grants that application access to the message. The IPC system is responsible for
+identifying which process has that application ID (if any) and giving the
+message to that process.
```

Comment:
```suggestion
Application IDs are generally used as a way to grant access to something. For
example, a process that wants to send a message to another process will
generally do so by sending the message to that process' application ID. Doing so
grants that application access to the message. The IPC system is responsible for
identifying which process has that application ID (if any) and giving the
message to that process.
```

I think this makes more sense using "Application ID" as the noun, rather than just applications, since applications being a set of processes with same AppID doesn't really grant access to anything.